### PR TITLE
Rackspace Load Balancer - simple edits

### PR DESCRIPTION
**Pull request by mynnx**
Enable PUTs on `/loadbalancer/<id>`, as documented here:

http://docs-beta.rackspace.com/loadbalancers/api/v1.0/clb-devguide/content/List_Load_Balancer_Details-d1e1522.html

### CODE REVIEW

File: libcloud/loadbalancer/drivers/rackspace.py [hunk 1]
```diff
@@ -292,6 +292,45 @@ def balancer_list_members(self, balancer):
...
+        @type       metadata: C{str}
+
+        @keyword    port: New load balancer port
+        @type       metadata: C{int}
+        """
+        attrs = self._to_update_attrs(kwargs)
+        resp = self.connection.request('/loadbalancers/%s' % balancer.id,
+                method='PUT',
+                data=json.dumps(attrs))
+        return resp.status == 202
```
Comment (Kami) [inline libcloud/loadbalancer/drivers/rackspace.py:324]: `== httplib.ACCEPTED`

File: libcloud/loadbalancer/drivers/rackspace.py [hunk 2]
```diff
@@ -292,6 +292,45 @@ def balancer_list_members(self, balancer):
...
+        update_attrs = {}
+        if "name" in attrs:
+            update_attrs['name'] = attrs['name']
+
+        if "algorithm" in attrs:
+            algorithm_value = self._algorithm_to_value(attrs['algorithm'])
+            update_attrs['algorithm'] = algorithm_value
+
+        if "protocol" in attrs:
+            update_attrs['protocol'] = attrs['protocol']
```
Comment (Kami) [inline libcloud/loadbalancer/drivers/rackspace.py]: create function does `protocol.upper()` so we should do the same thing here.

Ideally we would isolate this into a function.

File: libcloud/loadbalancer/drivers/rackspace.py (hunk 1 above)
Comment (mynnx) [inline libcloud/loadbalancer/drivers/rackspace.py:324]: 8a01c5e

File: libcloud/loadbalancer/drivers/rackspace.py (hunk 2 above)
Comment (mynnx) [inline libcloud/loadbalancer/drivers/rackspace.py]: 13fe9e4

File: libcloud/loadbalancer/drivers/rackspace.py [hunk 3]
```diff
@@ -285,13 +286,52 @@ def balancer_detach_member(self, balancer, member):
...
 
-        return resp.status == 202
+        return resp.status == httplib.ACCEPTED
 
     def balancer_list_members(self, balancer):
         uri = '/loadbalancers/%s/nodes' % (balancer.id)
         return self._to_members(
                 self.connection.request(uri).object)
 
+    def update_balancer(self, balancer, **kwargs):
```
Comment (Kami) [inline libcloud/loadbalancer/drivers/rackspace.py:310]: This method should go to the base class. It should also follow other API's (e.g. DNS) and return a modified `LoadBalancer` object.

Comment (mynnx) [inline libcloud/loadbalancer/drivers/rackspace.py:310]: c22f8f65, a90472b1, and 8d627e2c.  thanks for the feedback, @Kami!

File: libcloud/loadbalancer/base.py [hunk 4]
```diff
@@ -163,6 +163,25 @@ def get_balancer(self, balancer_id):
         raise NotImplementedError(
                 'get_balancer not implemented for this driver')
 
+    def update_balancer(self, balancer, **kwargs):
+        """
```
Comment (Kami) [inline libcloud/loadbalancer/base.py:167]: Need to update the docstring...

Comment (mynnx) [inline libcloud/loadbalancer/base.py:167]: fcb37c3a

File: libcloud/loadbalancer/drivers/rackspace.py (hunk 3 above)
Comment (Kami) [inline libcloud/loadbalancer/drivers/rackspace.py:310]: Update call is asynchronous which means currently solution won't really work (you will most likely get a non-modified object back).

One solution is to use a `PollingConnection` class. This class periodically polls and blocks until a job has completed. For more examples see Rackspace DNS driver.

File: libcloud/common/base.py [hunk 5]
```diff
@@ -618,14 +618,16 @@ def async_request(self, action, params=None, data='', headers=None,
...
+                                              context=context,
+                                              request_kwargs=kwargs)
 
         end = time.time() + self.timeout
         completed = False
         while time.time() < end and not completed:
             response = request(**kwargs)
             completed = self.has_completed(response=response)
-            time.sleep(self.poll_interval)
+            if not completed:
```
Comment (mynnx) [inline libcloud/common/base.py:629]: Also, we don't need to sleep if we're done polling.  This should speed up all `async_request`s.

File: libcloud/loadbalancer/drivers/rackspace.py (hunk 3 above)
Comment (mynnx) [inline libcloud/loadbalancer/drivers/rackspace.py:310]: Check it out now.  I had to tweak `PollingConnection` slightly, but I like this solution.  I also added an `ex_update_balancer_no_poll()`.

**Comment by Kami** [comment]
Merged in r1222444 .

Thanks.

**Pull request closed without merging**